core/ipld/codec/cbor/cbor.go

```diff
@@ -25,8 +28,20 @@ func (codec) Decode(b []byte, bind any, typ schema.Type, opts ...bindnode.Option
 
 var Codec = codec{}
 
-func Encode(val any, typ schema.Type, opts ...bindnode.Option) ([]byte, error) {
-	return ipld.Marshal(dagcbor.Encode, val, typ, opts...)
+func Encode(val any, typ schema.Type, opts ...bindnode.Option) (bytes []byte, err error) {
+	defer func() {
```

Comment:
nit: preference for switch on types here, e.g
```go
switch v := r.(type) {
  case error:
    // blah blah blah
```
might also be nice to grab the stack via `debug.Stack()`